src/blog/2024/10/how-to-build-automate-devops-pipelines-node-red-deployments.md

```diff
@@ -0,0 +1,88 @@
+_Add stages to your pipeline for different deployment environments, such as development, testing, and production_
+
+6. In the window that opens, select the **stage type** based on whether it's an instance, device, or device group.
+7. Enter the name for the stage in the **Stage Name** field.
+8. Choose an instance, device, or device group for the stage.
+9. Next, configure which action should be performed when this stage is pushed to the next:
+   - **Create New Snapshot:** Generates a new [Snapshot](/docs/user/high-availability/) using the current flows and settings.
+   - **Use Latest Instance Snapshot:** Uses the most recent existing snapshot of the instance. The deployment will fail if no snapshot exists.
+   - **Prompt to Select Snapshot:** Prompts at deploy time to select which snapshot from the source stage should be copied to the next stage.
+10. Check the option **Deploy to Devices** if you want changes to be deployed to all devices connected to this stage’s instance when the stage is deployed.
+
+![Screenshot showing the stage configuration form with options to select instance types and configure actions.](./images/form-to-create-configure-stages.png){data-zoomable}
+_Configure each stage by selecting an instance, device, or device group, and define the deployment actions_
+
+Once you’ve created your initial stage, you can add more stages by following the same process. This flexibility allows you to tailor your DevOps pipeline to meet the specific needs of your Node-RED deployment.
+
+For example, in development, you might have a Node-RED instance in the cloud to build your application. During staging, you could test the setup with a single device. Finally, in production, you can deploy the tested application to thousands of devices in a device group, saving time and ensuring smooth deployment at scale.
+
+![DevOps pipelines animation](./images/devops-pipeline.gif){data-zoomable}
+*Image: DevOps animation showing application deployment to production securely.*
+
+## Running a Pipeline Stage
+
+Once your pipeline is set up, you are ready to run it and deploy your changes across each stage. Here's how to do it:
+
+![Screenshot of the 'Run Pipeline' button in FlowFuse, allowing users to trigger the deployment process.](./images/button-to-run-pipeline.png){data-zoomable}
+_Click the 'Run Pipeline' button to initiate the automated deployment process across all configured stages_
+
+1. Click the run pipeline stage option. For each stage, this button will be available, except for the last one. Simply click it to start the deployment process.
+2. Once initiated, the deployment will automatically progress from the leftmost stage to the rightmost stage. It will copy all settings, environments, and flows and deploy them to the next stage. However, for environment variables, only the keys will be copied—you will need to manually add the values for each.
```

Comment:
This is not accurate. Each 2 adjacent stages operate independently

i.e. stating "the deployment will automatically progress from the leftmost stage to the rightmost stage" infers pressing the leftmost play button will operate the pipeline all the way to the rightmost stage


Additionally, the part about env var keys is misleading.

It goes something like: "The pipeline operation will copy env vars but wont overwrite any that exist in the target."

See this: 

![image](https://github.com/user-attachments/assets/43ff095f-a642-4a4e-8c27-7e7e0e21179a)